lib/provider.js

```diff
@@ -168,47 +176,39 @@ var transactionMethods = [
   'personal_sendTransaction'
 ];
 
-// helper list of RPC methods which execute code regardless of what their response object looks like
-var executingMethods = [
-  'eth_sendTransaction',
-  'eth_sendRawTransaction',
-  'eth_call',
-  'eth_estimateGas',
-  'debug_traceTransaction'
-];
-
 Provider.prototype._isTransactionRequest = function(request) {
   return transactionMethods.indexOf(request.method) != -1;
 }
 
-Provider.prototype._isExecutingRequest = function(request) {
-  return executingMethods.indexOf(request.method) != -1;
-}
-
 Provider.prototype.reportErrorInResponse = function(request, err, response) {
   const self = this;
   var newResponse = clone(response);
 
   if (!err) return newResponse;
 
-  // make sure we always return the transaction hash on failed transactions so
-  // the caller can get their tx receipt
-  if (self._isTransactionRequest(request) && err.hashes) {
-    newResponse.result = err.hashes[0];
+  // TODO: for next major release: move reporting of tx hash on error to error
+  // field to prevent poorly-written clients which assume that the existence of
+  // the "result" field implies no errors from breaking.
```

Comment:
I've witnessed this with web3 1.0 - when both `error` and `result` were set on the response sometimes it'd interpret the error correctly and sometimes it wouldn't.